extensions/blocks/opentable/opentable.php

```diff
@@ -0,0 +1,169 @@
+		if ( in_array( $attribute_name, array( 'iframe', 'newtab' ), true ) ) {
+			return $attributes[ $attribute_name ] ? 'true' : 'false';
+		}
+		return $attributes[ $attribute_name ];
+	}
+
+	$default_attributes = array(
+		'style'  => 'standard',
+		'iframe' => 'true',
+		'domain' => 'com',
+		'lang'   => 'en-US',
+		'newtab' => 'false',
+	);
+
+	return isset( $default_attributes[ $attribute_name ] ) ? $default_attributes[ $attribute_name ] : null;
+}
+
+/**
+ * Get the block type attribute
+ *
+ * @param array $attributes Array of block attributes.
+ *
+ * @return string The filtered attribute
+ */
+function get_type_attribute( $attributes ) {
+	if ( ! empty( $attributes['rid'] ) && count( $attributes['rid'] ) > 1 ) {
+		return 'multi';
+	}
+
+	if ( empty( $attributes['style'] ) || 'button' !== $attributes['style'] ) {
+		return 'standard';
+	}
+
+	return 'button';
+}
+
+/**
+ * Get the block theme attribute
+ *
+ * OpenTable has a confusing mix of themes and types for the widget. A type
+ * can have a theme, but the button style can not have a theme. The other two
+ * types (multi and standard) can have one of the three themes.
+ *
+ * We have combined these into a `style` attribute as really there are 4 styles
+ * standard, wide, tall, and button. Multi can be determined by the number of
+ * restaurant IDs we have.
+ *
+ * This function along with `jetpack_opentable_block_get_type_attribute`, translates
+ * the style attribute to a type and theme.
+ *
+ * Type        Theme      Style
+ * ==========|==========|==========
+ * Multi     |          |
+ * Standard  | Standard | Standard
+ *           | Wide     | Wide
+ *           | Tall     | Tall
+ * Button    | Standard | Button
+ *
+ * @param array $attributes Array of block attributes.
+ *
+ * @return string The filtered attribute
+ */
+function get_theme_attribute( $attributes ) {
+	$valid_themes = array( 'standard', 'wide', 'tall' );
+
+	if ( empty( $attributes['style'] )
+		|| ! in_array( $attributes['style'], $valid_themes, true )
+		|| 'button' === $attributes['style'] ) {
+		return 'standard';
+	}
+
+	return $attributes['style'];
+}
+
+/**
+ * Build an embed URL from an array of block attributes.
+ *
+ * @param array $attributes Array of block attributess.
+ *
+ * @return string Embed URL
+ */
+function build_embed_url( $attributes ) {
+	$url = add_query_arg(
+		array(
+			'type'   => get_type_attribute( $attributes ),
+			'theme'  => get_theme_attribute( $attributes ),
+			'iframe' => get_attribute( $attributes, 'iframe' ),
+			'domain' => get_attribute( $attributes, 'domain' ),
+			'lang'   => get_attribute( $attributes, 'lang' ),
+			'newtab' => get_attribute( $attributes, 'newtab' ),
+		),
+		'//www.opentable.com/widget/reservation/loader'
+	);
+
+	if ( ! empty( $attributes['rid'] ) ) {
+		foreach ( $attributes['rid'] as $rid ) {
+			$url .= '&rid=' . $rid;
+		}
+	}
+
+	return apply_filters( 'jetpack_opentable_block_url', esc_url( $url ) );
```

Comment:
Since we're `esc_url` at the very end, as we add the URL here, maybe we don't need it here anymore?
https://github.com/Automattic/jetpack/blob/add/opentable/extensions/blocks/opentable/opentable.php#L53

It's also a good opportunity to add a docblock for that filter, so our parser can generate a page for that filter in the Jetpack Codex.

I would also recommend passing the attributes to the filter, since one could use them when modifying the final URL.

```suggestion
	/**
	 * Filter the OpenTable URL used to embed a widget.
	 *
	 * @since 8.2.0
	 *
	 * @param string $url        OpenTable embed URL.
	 * @param array  $attributes Array of block attributes.
	 */
	return apply_filters( 'jetpack_opentable_block_url', $url, $attributes );
```

Reply:
That's much better, thanks!